src/config/statsd.ts

```diff
@@ -35,6 +38,22 @@ function hrtimer() {
   };
 }
 
+export const elapsedTimeMetrics = (path: string) => {
+  const expressStatsdLogger = bunyan.createLogger({ name: 'elapsedTimeInMs' });
+  const elapsedTimeInMs = hrtimer();
```

Comment:
this will only get the difference in time from the start of the app to when each function is called, not giving an accurate elapsed time.

Reply:
Do you have a suggestion on how to fix it?

Reply:
```
const expressStatsdLogger = bunyan.createLogger({ name: 'elapsedTimeInMs' });
export const elapsedTimeMetrics = (req: Request, res: Response, next: NextFunction) => {
    const elapsedTimeInMs = hrtimer();
    const path = req.path;
    const tags = { path, method: req.method };

    res.on('finish', () => {
      expressStatsdLogger.info(`${path} : ${elapsedTimeInMs()}`);
      statsd.histogram('elapsedTimeInMs', elapsedTimeInMs(), tags);
    });

    next();
};
```
And you can use it by doing `app.get(authFunction, elapsedTimeMetrics, yourGetFunction);` It has the be right before the main function itself.

Reply:
Oh nice. Thanks for this and the `once` tip.